compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -1890,29 +1890,31 @@ object Types {
     }
 
     /** The signature of the last known denotation, or if there is none, the
-     *  signature of the symbol
+     *  signature of the symbol. Signatures are always computed before erasure, since
+     *  some symbols change their signature at erasure.
      */
-    protected def computeSignature(implicit ctx: Context): Signature = {
+    protected def computeSignature(implicit ctx: Context): Signature =
       val lastd = lastDenotation
-      if (lastd != null) lastd.signature
+      val isErased = ctx.erasedTypes
+      if lastd != null && !isErased then lastd.signature
```

Comment:
`ctx.erasedTypes` being false does not mean that `lastd` itself is before erasure, for that one would have to check `last.validFor.firstPhaseId < ctx.erasurePhase.id` I think (this might be off by one).

Reply:
Yes, good point. I have fixed it now.